lib/redactable.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+# A module that adds redacted functionality to a class
+module Redactable
+  def self.included(mod)
+    class << mod
+      attr_reader :allowed_members
+    end
+  end
+
+  def inspect
+    name_or_nil = self.class.name ? " #{self.class.name}" : nil
+
+    attributes = members.map do |member|
+      members_h = to_h
```

Comment:
We probably want this `to_h` on the outside so we're not re-creating it for every single key
```suggestion
    members_h = to_h

    attributes = members.map do |member|
```